src/metabase/models/collection.clj

```diff
@@ -68,18 +68,25 @@
 
 (defmethod mi/can-write? Collection
   ([instance]
-   (if (= (:entity_id instance) (perms/default-audit-collection-entity-id))
+   (mi/can-write? :model/Collection (:id instance)))
+  ([model pk]
+   (if (= pk (perms/default-audit-collection-id))
      false
-     (mi/current-user-has-full-permissions? :write instance)))
-  ([_ pk]
-   (mi/can-write? (t2/select-one :model/Collection :id pk))))
+     (mi/current-user-has-full-permissions? :write model pk))))
 
 (defmethod mi/can-read? Collection
   ([instance]
    (perms/can-read-audit-helper :model/Collection instance))
   ([_ pk]
    (mi/can-read? (t2/select-one :model/Collection :id pk))))
 
+(defn collection-entity-id->id
+  "Returns id from entity id for collections. Memoizes from entity id."
+  [entity-id]
+  ((mdb.connection/memoize-for-application-db
+    (fn [entity-id]
+      (t2/select-one-pk :model/Collection :entity_id entity-id))) entity-id))
+
```

Comment:
Not sure I love having this as a generic collections helper since it's pretty audit-v2 specific. Unless there are other use cases in mind it might make more sense to put it next to its callers in the EE code